src/app/components/client/StripeScript.tsx

```diff
@@ -0,0 +1,13 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
```

Comment:
Could you explain why this needs to be a client component, and why it can't just be imported directly in the layout?

Reply:
I opted for including `stripe.js` [as a module](https://docs.stripe.com/js/including), this is also how [MDN does it as well](https://github.com/mdn/yari/blob/2f4a09d0021817207f1dc3a1dea368267ab49d28/client/src/plus/offer-overview/offer-overview-subscribe/stripe.tsx).

An alternative would be to directly add `<script src="https://js.stripe.com/v3" async></script>`.  When it’s [imported as a side effect](https://github.com/stripe/stripe-js?tab=readme-ov-file#import-as-a-side-effect), like it is in this PR, the `<script>` is being [injected client side](https://github.com/stripe/stripe-js/blob/master/src/shared.ts#L38) and expects the DOM to be present — thus the `"use client"`.

Reply:
But client components are still pre-rendered on the server, so there might still not be a DOM present. Possibly you need to do a dynamic `import("@stripe/stripe-js")` inside a `useEffect` to make it only run on the client side?